lambdas/db-migration/src/index.ts

```diff
@@ -0,0 +1,52 @@
+import Knex from 'knex';
+import * as path from 'path';
+
+export type Command = 'latest';
+
+export interface HandlerEvent {
+  command?: Command,
+  env?: NodeJS.ProcessEnv
+}
+
+const getRequiredEnvVar = (name: string, env: NodeJS.ProcessEnv): string => {
+  const value = env?.[name];
+
+  if (value) return value;
+
+  throw new Error(`The ${name} environment variable must be set`);
+};
+
+const getConnectionConfig = (env: NodeJS.ProcessEnv): Knex.PgConnectionConfig => ({
+  host: getRequiredEnvVar('PG_HOST', env),
+  user: getRequiredEnvVar('PG_USER', env),
+  // TODO Get this value from secrets manager
+  password: getRequiredEnvVar('PG_PASSWORD', env)
+});
+
+export const handler = async (event: HandlerEvent): Promise<void> => {
+  const env = event?.env ?? process.env;
+
+  const knex = Knex({
+    client: 'pg',
+    connection: getConnectionConfig(env),
+    debug: env?.KNEX_DEBUG === 'true',
+    asyncStackTraces: env?.KNEX_ASYNC_STACK_TRACES === 'true',
+    migrations: {
+      directory: path.join(__dirname, 'migrations')
+    }
+  });
+
+  const command = event?.command ?? 'latest';
+
+  try {
+    switch (command) {
+      case 'latest':
+        await knex.migrate.latest();
+        break;
+      default:
+        throw new Error(`Invalid command: ${command}`);
+    }
+  } finally {
```

Comment:
I haven't tested knex's behavior here, but does knex always do the right thing here if it's using transactions in the migration (e.g. a failure to run all of the migrations in sequence results in a full rollback attempt?)

Reply:
From [here](http://knexjs.org/#Migrations):

> By default, each migration is run inside a transaction.

I'm not sure what happens if a migration succeeds and the next migration fails. That'll be worth testing.

Reply:
Definitely - obviously if it *doesn't* do the right thing we should consider implementing a catch with an explicit rollback logic.    Not sure if that should be pushed off as scope creep or not, I'm fine with it being in <the next ticket> or YAGNIng it for now, so long as we don't let it go and lose track of the concern.